protect against download_url hijacking

JarbasAl opened the pull request.

ChanceNCounter reviewed (commented):
I'm seeing 16 other references to this element in code, incl:

`skill_entry.py`
`github/__init__.py`
`github/api.py`
`github/raw.py`

I'm only driving by. Does this need to be addressed?

JarbasAl commented:
i think i addressed all the relevant places, at worst this could cause log pollution 

i left the places where i extracted the download_url commented out but in there in case we want to revisit, but i think i didn't miss any place where this url was injected in parsed data (only happened when parsing releases i think)

ChanceNCounter reviewed (changes requested):
Still some occurrences of search term `["download` in api.py. Otherwise, just remains to test. Definitely gotta put up those test repos, that was a good call 😁
